website/addons/s3/tests/test_view.py

```diff
@@ -219,185 +121,63 @@ def test_user_settings_cant_list(self, mock_can_list):
         assert_equals(rv.status_int, http.BAD_REQUEST)
         assert_in('Unable to list buckets.', rv.body)
 
-    @mock.patch('website.addons.s3.views.config.utils.can_list', return_value=True)
-    def test_node_settings_no_user_settings(self, mock_can_list):
-        self.node_settings.user_settings = None
-        self.node_settings.save()
-        url = self.project.api_url_for('s3_authorize_node')
-
-        self.app.post_json(url, {'access_key': 'scout', 'secret_key': 'ssshhhhhhhhh'}, auth=self.user.auth)
-        self.user_settings.reload()
-        assert_equals(self.user_settings.access_key, 'scout')
-
-    @mock.patch('website.addons.s3.views.config.utils.get_bucket_names')
-    def test_s3_bucket_list(self, mock_bucket_list):
-        fake_buckets = []
-        for _ in range(10):
-            fake_bucket = mock.Mock()
-            fake_bucket.name = fake.domain_word()
-            fake_bucket.append(fake_bucket)
-
-        mock_bucket_list.return_value = fake_buckets
-        url = self.node_settings.owner.api_url_for('s3_get_bucket_list')
-        ret = self.app.get(url, auth=self.user.auth)
-
-        assert_equals(ret.json, {'buckets': [bucket.name for bucket in fake_buckets]})
-
     def test_s3_remove_node_settings_owner(self):
-        url = self.node_settings.owner.api_url_for('s3_delete_node_settings')
+        url = self.node_settings.owner.api_url_for('s3_deauthorize_node')
         ret = self.app.delete(url, auth=self.user.auth)
-
-        assert_equal(ret.json['has_bucket'], False)
-        assert_equal(ret.json['node_has_auth'], False)
+        result = self.Serializer().serialize_settings(node_settings=self.node_settings, current_user=self.user)
+        assert_equal(result['hasBucket'], False)
```

Comment:
Because that view function no longer returns a JSON response, as it does not need to. `S3Serializer.serialize_settings` does effectively what the old `AddonS3NodeSettings.to_json` did.